### PR TITLE
[WIP] 2.0.0-dev.1.2

### PR DESCRIPTION
**Pull request by lexaknyazev**
This point version update is focused on polishing wrappers. 
I'm going to update drag-n-drop front-end too before merging.

/cc @emackey @donmccurdy

### CODE REVIEW

File: tool/npm_template/node_wrapper.dart [hunk 1]
```diff
@@ -68,29 +63,32 @@ abstract class JSValidationOptions {
...
 void main() {
-  exports.validateBytes = allowInterop((String filename, Uint8List data,
-          GetResourceCallback getResource, JSValidationOptions options) =>
+  exports.validateBytes = allowInterop((Uint8List data, Object options) =>
       new Promise<Object>(allowInterop((resolve, reject) {
-        validateBytes(filename, data, getResource, options).then((result) {
+        validateBytes(data, options).then((result) {
           resolve(jsify(result));
-        }, onError: reject);
+        }, onError: (Object e) => reject(e.toString()));
```
Comment (emackey) [inline tool/npm_template/node_wrapper.dart]: Does the client lose access to anything important as a result of hard-coding the `toString` here?

Comment (emackey) [reply tool/npm_template/node_wrapper.dart]: For example maybe a stack trace from the non-minified version of this?

Comment (lexaknyazev) [reply tool/npm_template/node_wrapper.dart]: All properties of that object will be minified, so they won't be really accessible. I'll look into how to provide a stack trace.

**Comment by lexaknyazev** [comment]
@emackey See updated rejection handling.

File: tool/npm_template/node_wrapper.dart [hunk 2]
```diff
@@ -66,18 +66,24 @@ external Exports get exports;
 external List<String> _getKeys(Object value);
 
 void main() {
+  const kDebug =
+      const bool.fromEnvironment('GLTF_VALIDATOR_DEBUG', defaultValue: false);
+
+  void onError(Object e, StackTrace st, Function reject) =>
+      kDebug ? reject(st.toString()) : reject(e.toString());
```
Comment (emackey) [inline tool/npm_template/node_wrapper.dart]: Sorry to nitpick, one last suggestion here.  The debug mode looks like it hides the actual error message in the process of providing a stack trace.  I think both would be needed.

```
    kDebug ? reject(e.toString() + '\n' + st.toString()) : reject(e.toString());
```

File: tool/npm_template/index.js [hunk 3]
```diff
@@ -17,36 +17,32 @@
 const validator = require('./gltf_validator.dart.js');
 
 /**
- * @callback loadExternalResourceCallback
- * @param {string} uri - Relative URI of the external resource
- * @returns {Promise} - Promise with Uint8Array data
+ * @callback ExternalResourceFunction
```
Comment (emackey) [inline tool/npm_template/index.js]: Lowercase `e`

Comment (emackey) [reply tool/npm_template/index.js]: Oh I think I'm wrong about that

File: tool/npm_template/node_wrapper.dart (hunk 2 above)
Comment (lexaknyazev) [reply tool/npm_template/node_wrapper.dart]: StackTrace string representation includes error message.

Comment (emackey) [reply tool/npm_template/node_wrapper.dart]: Ah OK

**Comment by emackey** [comment]
Published `gltf-validator@2.0.0-dev.1.2`.

**Pull request merged**